### PR TITLE
Rename ServeWithListenerStopped to Serve in secure_serving

### PR DESCRIPTION
**Pull request by ardaguclu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This PR removes Serve function and uses all required places
ServeWithListenerStopped which takes place new Serve function.

This function returns ListenerStopped channel can be used to drain
requests before shutting down the server.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

### CODE REVIEW

**Comment by ardaguclu** [comment]
cc: @tkashem

**Review by damemi — COMMENTED** [review]
/approve
for scheduler change

File: staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go [hunk 1]
```diff
@@ -131,16 +131,21 @@ func (s *Server) Start(ctx context.Context) error {
...
 		var err error
-		shutdownCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
+		shutdownCh, listenerStoppedCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
 		if err != nil {
 			return fmt.Errorf("failed to start secure server: %w", err)
 		}
 	}
 
+	<-listenerStoppedCh
+	klog.V(1).InfoS("[graceful-termination] HTTP Server is stopped listening")
```
Comment (tkashem) [inline staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go:144]: nit: `HTTP Server has stopped listening`

**Review by tkashem — COMMENTED** [review]
looks good, have one minor comment

File: staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go (hunk 1 above)
Comment (ardaguclu) [reply staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go]: Thanks @tkashem. Done.

**Comment by ardaguclu** [comment]
cc: @sttts

**Comment by ardaguclu** [comment]
@tkashem could you please have a look at this change?, thanks

File: staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go [hunk 2]
```diff
@@ -131,16 +131,21 @@ func (s *Server) Start(ctx context.Context) error {
...
 		var err error
-		shutdownCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
+		shutdownCh, listenerStoppedCh, err = s.secureServing.Serve(mux, 0, ctx.Done())
 		if err != nil {
 			return fmt.Errorf("failed to start secure server: %w", err)
 		}
 	}
 
+	<-listenerStoppedCh
+	klog.V(1).InfoS("[graceful-termination] HTTP Server has stopped listening")
```
Comment (liggitt) [inline staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go:144]: is this the right order? the listener stops before all the requests have completed being handled?

Comment (liggitt) [reply staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go:144]: it's not super clear to me why we would want to block exit after the requests have completed being handled (`<-shutdownCh`)

File: staging/src/k8s.io/apiserver/pkg/server/secure_serving.go [hunk 3]
```diff
@@ -142,73 +142,8 @@ func (s *SecureServingInfo) tlsConfig(stopCh <-chan struct{}) (*tls.Config, erro
...
-		}
-	}
-
-	// use tlsHandshakeErrorWriter to handle messages of tls handshake error
-	tlsErrorWriter := &tlsHandshakeErrorWriter{os.Stderr}
-	tlsErrorLogger := log.New(tlsErrorWriter, "", 0)
-	secureServer.ErrorLog = tlsErrorLogger
-
-	klog.Infof("Serving securely on %s", secureServer.Addr)
-	stoppedCh, _, err := RunServer(secureServer, s.Listener, shutdownTimeout, stopCh)
```
[the remark this answers is not in the document]
Comment (ardaguclu) [reply staging/src/k8s.io/apiserver/pkg/server/secure_serving.go:201]: Yes, they are identical. We are just returning `listenerStoppedCh` additionally which is ignored in here.

File: staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go (hunk 2 above)
Comment (sttts) [reply staging/src/k8s.io/pod-security-admission/cmd/webhook/server/server.go:144]: > is this the right order? the listener stops before all the requests have completed being handled?

Listener != server. So yes, we stop listening for new requests before in-flights requests are all finished.

**Pull request merged**